src/Forms/FormRequestHandler.php

```diff
@@ -228,23 +228,23 @@ public function httpSubmission($request)
             // First, try a handler method on the controller (has been checked for allowed_actions above already)
             $controller = $this->form->getController();
             if ($controller && $controller->hasMethod($funcName)) {
-                return $controller->$funcName($vars, $this->form, $request);
+                return $controller->$funcName($vars, $this->form, $request, $this);
```

Comment:
If you add a 4th argument to the request handler, please make sure to add it consistently to each of the other callbacks (there are 3 other places $funcName is called).